Add deprecation notices to operator (#1230)
kyma-project/serverless: docs/user/00-20-configure-serverless.md

       enableInternal: true
   ```

> [!CAUTION]

> > The `spec.dockerRegistry` field is deprecated and will be removed in a future version of Serverless when functions won't require building images.

Reviewer: ```suggestion
> The `spec.dockerRegistry` field is deprecated and will be removed in a future version of Serverless when Functions won't require building images.
```

      targetCPUUtilizationPercentage: 50
```

> [!CAUTION]

> > The `spec.targetCPUUtilizationPercentage` field is deprecated and will be removed in a future version of Serverless that disables automatic HPA creation.

Reviewer: ```suggestion
> The `spec.targetCPUUtilizationPercentage` field is deprecated and will be removed in a future version of Serverless, where automatic HPA creation will be disabled.
```

      functionBuildExecutorArgs: "--insecure,--skip-tls-verify,--skip-unused-stages,--log-format=text,--cache=true,--use-new-run,--compressed-caching=false"
```

> [!CAUTION]

> > The `spec.functionBuildExecutorArgs` field is deprecated and will be removed in a future version of Serverless when functions won't require building images.

Reviewer: ```suggestion
> The `spec.functionBuildExecutorArgs` field is deprecated and will be removed in a future version of Serverless when Functions won't require building images.
```

      functionBuildMaxSimultaneousJobs: 5
```

> [!CAUTION]

> > The `spec.functionBuildMaxSimultaneousJobs` field is deprecated and will be removed in a future version of Serverless when functions won't require building images.

Reviewer: ```suggestion
> The `spec.functionBuildMaxSimultaneousJobs` field is deprecated and will be removed in a future version of Serverless when Functions won't require building images.
```

      defaultBuildJobPreset: "normal"
```

> [!CAUTION]

> > The `spec.defaultBuildJobPreset` field is deprecated and will be removed in a future version of Serverless when functions won't require building images.

Reviewer: ```suggestion
> The `spec.defaultBuildJobPreset` field is deprecated and will be removed in a future version of Serverless when Functions won't require building images.
```
